web/pgadmin/tools/sqleditor/static/js/components/sections/ResultSet.jsx

```diff
@@ -1239,11 +1244,36 @@ export function ResultSet() {
     return ()=>eventBus.deregisterListener(QUERY_TOOL_EVENTS.TRIGGER_RENDER_GEOMETRIES, renderGeometries);
   }, [rows, columns, selectedRows.size, selectedColumns.size]);
 
-  const handleScroll = (e)=>{
+  const handleScroll = (e) => {
+    // Set scroll current position of RestSet.
+    if (!_.isNull(e.currentTarget) && isResetScroll.current) {
+      resultGridRef.current = { ...e };
+      scrollPosition.current = {
+        top: e.currentTarget.scrollTop,
+        left: e.currentTarget.scrollLeft
+      };
+    }
+
     if (isLoadingMore || !rsu.current.isAtBottom(e)) return;
     eventBus.fireEvent(QUERY_TOOL_EVENTS.FETCH_MORE_ROWS);
   };
 
+  const triggerResetScroll = () => {
+    // Reset the scroll position to previously saved location.
+    if (resultGridRef.current) {
+      isResetScroll.current = false;
```

Comment:
What is the use of `isResetScroll` ?

Reply:
It is required to prevent setting the default scroll on revisit.

Reply:
From what I understand, it can be renamed to `isResettingScroll`.